cli.js

```diff
@@ -1,20 +1,128 @@
 #!/usr/bin/env node
-'use strict'
-const spawnSync = require('child_process').spawnSync
-const COMPILER_PATH = require('./index.js').compiler.COMPILER_PATH
-const result = spawnSync(
-  'java',
-  ['-jar', COMPILER_PATH].concat(process.argv.slice(2)),
-  {stdio: 'inherit'}
-)
-if (result.error) {
-  if (result.error.code === 'ENOENT') {
-    console.error('Could not find "java" in your PATH.')
-  } else {
-    console.error(result.error.message)
-  }
-  process.exit(1)
+'use strict';
+const fs = require('fs');
+const {getNativeImagePath, getFirstSupportedPlatform, parseCliFlags} = require('./lib/utils');
+
+const compilerFlags = parseCliFlags(process.argv.slice(2));
+let platform;
+if (compilerFlags.platform) {
+  platform = compilerFlags.platform;
+  delete compilerFlags.platform;
```

Comment:
Note to self: this deletion pattern is consuming flags before the `new Compiler` later.